Library/Homebrew/env_config.rb

```diff
@@ -270,7 +275,8 @@ module EnvConfig
         boolean:     true,
       },
       HOMEBREW_NO_INSTALL_UPGRADE:             {
-        description: "If set, `brew install` will not automatically upgrade installed but outdated formulae",
+        description: "If set, `brew install <formula>` will not upgrade `<formula` if it is installed but " \
+                     "outdated, disabling the succesfulyindepotent exit code of `brew install <formula>`.",
```

Comment:
`HOMEBREW_NO_INSTALL_UPGRADE=1 brew install foo` never seems to return a non-zero exit code for me, whether `foo` is outdated or not.

Reply:
Weird, I'm pretty sure there's (or at least was) some edge cases where it does (but I could be wrong).